### ISSUE TITLE
Material sorting has changed?

### ISSUE THREAD

**Issue by SebastianLopienski** [indico/indico#2164]
I don't know when it changed - but previously these three materials were sorted this way (hashtag first, as `ord('#')` < `ord('A')`):
* #B
* A
* C

And now they are sorted differently, the hash being ignored:
* A
* #B
* C 

Is this desired, or just an accidental change? Will it stay this way? I certainly preferred the previous sorting :-(

**Comment by ThiefMaster** [comment]
Without checking deeper in the code, I think we use Postgres-level sorting so it *should* work fine...

```
 select u from unnest('{b,a,#b,c}'::varchar[]) u ORDER BY lower(u) ASC;
 u
----
 #b
 a
 b
 c
(4 rows)
```

Unless we missed an `ORDER BY` somewhere (in which case it would use whatever order Postgres uses to store the rows)...

**Comment by ThiefMaster** [comment]
ok, very strange. on our production database the order is different for the very same statement:

```
indico> select u from unnest('{b,a,#b,c}'::varchar[]) u ORDER BY lower(u) ASC;
+-----+
| u   |
|-----|
| a   |
| b   |
| #b  |
| c   |
+-----+
```

**Comment by SebastianLopienski** [comment]
You see how attentive your users are, to spot such subtleties :-)

**Comment by ThiefMaster** [comment]
```
indico> select u from unnest('{b,a,#b,c}'::varchar[]) u ORDER BY lower(u) COLLATE "en_US";
+-----+
| u   |
|-----|
| a   |
| b   |
| #b  |
| c   |
+-----+
SELECT 4
Time: 0.001s


indico> select u from unnest('{b,a,#b,c}'::varchar[]) u ORDER BY lower(u) COLLATE "C";
+-----+
| u   |
|-----|
| #b  |
| a   |
| b   |
| c   |
+-----+
SELECT 4
Time: 0.002s
```

Prod uses `en_US` collation, my dev db uses the `C` collation. So I'm not sure whether there's any proper way to solve this...

**Comment by pferreir** [comment]
> So I'm not sure whether there's any proper way to solve this...

Client-side (meaning, web server-side) sorting?

**Comment by ThiefMaster** [comment]
Yeah, but Python sorting is not awesome either (I think besides the 'special' handling of `#` and other symbols the locale-aware sorting in Postgres makes more sense):

``` python
>>> print u', '.join(sorted([u'a', u'b', u'c', u'#b', u'ä', u'A'], key=unicode.lower))
#b, a, A, b, c, ä
```

**Comment by pferreir** [comment]
Yeah, it should be something a little bit smarter, which un-accents unicode:

https://github.com/indico/indico/blob/master/indico/util/string.py#L67

**Comment by ThiefMaster** [comment]
I wonder if it's possibly to add custom sorting rules to Postgres.. couldn't find anything on a quick google search but considering how much things you can customize in Postgres it might be possible nonetheless

**Comment by SebastianLopienski** [comment]
Just a related comment: since "Recording" material is added afterwards, and automatically (not by the chairperson or speakers), then would it make sense to either distinguish it visually somehow (e.g. an icon?), or always list as the last material? Otherwise it's mixed with other materials, e.g. see here: https://indico.cern.ch/event/392796/

**Comment by ThiefMaster** [comment]
Recording links are just normal material links so they are sorted alphabetically - there are no special material types that would affect how they are displayed/sorted.

Theoretically we could use a custom icon based on the link target, but even then we'd need to use the same icon for all links pointing to CDS.

**Comment by SebastianLopienski** [comment]
Hi guys, any news on this material sort order issue?

**Comment by DirkHoffmann** [comment]
I would prefer the server side setup, as proposed by Pedro, with the present implementation. It does not even have to be system wide; a proper environment variable for the webserver uid should be sufficient.

But if material could be sorted in any way the user desires, that would be a nice feature as well.

**Comment by pferreir** [comment]
@SebastianLopienski We're busy with more urgent things, unfortunately. But let's see if we can sort this out before the next release.

**Comment by SebastianLopienski** [comment]
@pferreir OK, thanks.
